lmp-auto-hostname: improving user guide (#269)
foundriesio/docs: source/user-guide/lmp-auto-hostname/lmp-auto-hostname.rst

.. prompt:: bash host:~$

    git clone -b devel https://source.foundries.io/factories/<factory>/meta-subscriber-overrides.git
    cd meta-subscriber-overrides

Edit the ``recipes-samples/images/lmp-factory-image.bb`` file and add the recipe on the ``CORE_IMAGE_BASE_INSTALL`` list:

.. prompt:: bash host:~$, auto

    host:~$ gedit recipes-samples/images/lmp-factory-image.bb

**recipes-samples/images/lmp-factory-image.bb**:

.. prompt:: text

     diff --git a/recipes-samples/images/lmp-factory-image.bb b/recipes-samples/images/lmp-factory-image.bb
     --- a/recipes-samples/images/lmp-factory-image.bb
     +++ b/recipes-samples/images/lmp-factory-image.bb
     @@ -30,6 +30,7 @@ CORE_IMAGE_BASE_INSTALL += " \
          networkmanager-nmcli \
          git \
          vim \
     +    lmp-auto-hostname \
          packagegroup-core-full-cmdline-extended \
          ${@bb.utils.contains('LMP_DISABLE_GPLV3', '1', '', '${CORE_IMAGE_BASE_INSTALL_GPLV3}', d)} \
     "

LmP Auto Hostname Variables
^^^^^^^^^^^^^^^^^^^^^^^^^^^

The ``lmp-auto-hostname`` recipe can be configured through variables.

.. confval:: LMP_HOSTNAME_MODE=<option>
    :default: ``serial``

    .. option:: serial

>        appends the serial number of the device.

Reviewer: I'm not sure I understand which serial number is it. Would it be some FF serial number? Or the SoC serial number?

Author: https://github.com/foundriesio/meta-lmp/blob/master/meta-lmp-base/recipes-support/lmp-auto-hostname/lmp-auto-hostname/lmp-update-hostname.sh#L23

I think at this point if the customer wants more detail, he needs to take a look at the reference manual or read the recipe.
Don't want to extend too much the UG.

          host:~$ git commit -m "lmp-auto-hostname: Adding recipe" recipes-samples/images/lmp-factory-image.bb
          host:~$ git push

   .. group-tab:: MAC

      Edit the ``conf/machine/include/lmp-factory-custom.inc`` file and add the variables:
      
      .. prompt:: bash host:~$, auto
      
          host:~$ gedit recipes-samples/images/lmp-factory-image.bb
      
      **conf/machine/include/lmp-factory-custom.inc**:
      
      .. prompt:: text
      
           LMP_HOSTNAME_MODE = "mac"
           LMP_HOSTNAME_NETDEVICE = "eth0"
      
      Add the changed files, commit and push:

      .. prompt:: bash host:~$, auto

          host:~$ git add recipes-samples/images/lmp-factory-image.bb
          host:~$ git add conf/machine/include/lmp-factory-custom.inc
          host:~$ git commit -m "lmp-auto-hostname: Adding recipe"
          host:~$ git push

The latest **Target** named ``platform-devel`` should be the CI job you just created.

When FoundriesFactory CI finishes all jobs, if your device is already registered, 
wait until the Over-the-Air update finishes, otherwise download and flash the image.

Testing Auto Hostname
^^^^^^^^^^^^^^^^^^^^^

> Log in to the device via SSH and it will be possible to see right after ``fio@`` the new hostname.

Reviewer 2: ```suggestion
Log in to the device via SSH and check the new hostname right after ``fio@``.
```

Check also the file ``/etc/hostname`` to confirm the new hostname.

.. tabs::

   .. group-tab:: Serial
      
      .. prompt:: bash fio@raspberrypi3-64-51ca7875:~$
      
          cat /etc/hostname 
      
      **Example Output**:
      
      .. prompt:: text
      
           raspberrypi3-64-51ca7875
      
   .. group-tab:: MAC
      
      .. prompt:: bash fio@raspberrypi3-64-b827ebca7875:~$
      
          cat /etc/hostname 
      
      **Example Output**:
      
      .. prompt:: text
      
           raspberrypi3-64-b827ebca7875

Summary
^^^^^^^

This user guide gives you an example of how to enable the LmP Auto Hostname. 

> It helps when each device needs a custom hostname.

Reviewer: I would drop the summary for the UG pages.
